tests/suites/helpers.function

```diff
@@ -373,12 +374,13 @@ typedef struct
     uint8_t expected_call;
     uint8_t expected_call_happened;
 
+    jmp_buf state;
+
     mbedtls_test_param_failed_location_record_t location_record;
 }
 param_failed_ctx_t;
 static param_failed_ctx_t param_failed_ctx;
 
-jmp_buf param_fail_jmp;
 jmp_buf jmp_tmp;
```

Comment:
I haven't changed anything around `jmp_tmp`. It is necessary because when MBEDTLS_CHECK_PARAMS is enabled and within a test case TEST_INVALID_PARAM is called there are two long jumps set-up at the same time: the one in main_test.function: execute_function_ptr and the one in TEST_INVALID_PARAM  macro.